modules/project-factory/factory-projects.tf

```diff
@@ -39,15 +39,15 @@ locals {
     for k, v in local._projects : [
       for b in try(v.billing_budgets, []) : {
         budget  = b
-        project = k
+        project = lookup(v, "name", k)
       }
     ]
   ])
   project_budgets = {
     for v in local._project_budgets : v.budget => v.project...
   }
   projects = {
-    for k, v in local._projects : k => merge(v, {
+    for k, v in local._projects : lookup(v, "name", k) => merge(v, {
```

Comment:
Should the filename or name win as a key?

Reply:
if name present in yaml, use that, otherwise filename

Reply:
seems counterintuitive, but you came up with the use case so 👍